data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/config/LogDynamicFileAppenderConfig.java

```diff
@@ -0,0 +1,56 @@
+package gov.cdc.etldatapipeline.organization.config;
+
+import ch.qos.logback.core.FileAppender;
+import lombok.Setter;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+@Setter
+public class LogDynamicFileAppenderConfig<E> extends FileAppender<E> {
+
+    private String logFilePath;
+
+    /**
+     * Purpose: Dynamically create log file if not exist
+     */
+    @Override
+    public void start() {
+        if (logFilePath == null) {
+            addError("Log file path is not configured");
+            return;
+        }
+
+        if (logFilePath.contains("%d{")) {
+            SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd_HH-mm-ss");
+            dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
```

Comment:
Even though this is a logging module, do we keep this in UTC?

Reply:
Yeah we can change it to local timezone if required.